### PR TITLE
Add apt_repo module

### PR DESCRIPTION
**Pull request by Obirvalger**
##### SUMMARY
Add module apt_repo to work with apt repositories in ALT based distributions.

The apt_repo module allows working with apt repositories via tool apt-repo (https://www.altlinux.org/Apt-repo).

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
apt_repo

### CODE REVIEW

File: lib/ansible/modules/packaging/os/apt_repo.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+description:
+  - Manages APT repositories using apt-repo tool.
+  - See U(https://www.altlinux.org/Apt-repo) for details about apt-repo
+notes:
+    - This module works on ALT based distros.
+version_added: "2.8"
+options:
+  repo:
+    description:
+      - Name of package to install, upgrade or remove.
```
Comment (gundalow) [inline lib/ansible/modules/packaging/os/apt_repo.py]:
```suggestion
      - Name of the package to install, upgrade or remove.
      - The C(all) package can be used to ....
```

File: lib/ansible/modules/packaging/os/apt_repo.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    apt_repo(module, 'rm', 'all')
+    apt_repo(module, 'add', repo)
+
+
+def update(module):
+    """update package cache"""
+    apt_repo(module, 'update')
+
+
+def main():
```
Comment (gundalow) [inline lib/ansible/modules/packaging/os/apt_repo.py:114]: What would be needed to support "check mode"?

Comment (Obirvalger) [reply lib/ansible/modules/packaging/os/apt_repo.py:114]: Need to add such functionality to apt-repo tool.
This work is in progress.

Comment (gundalow) [reply lib/ansible/modules/packaging/os/apt_repo.py:114]: @Obirvalger How would you like to progress?
Mark this PR as `WIP` (work in progress) till apt-repo tool supports this functionality
Merge this PR as-is, and update the module at a later date once apt-repo tool has been updated?

Comment (Obirvalger) [reply lib/ansible/modules/packaging/os/apt_repo.py:114]: I would like to add as-is, and update later.

File: lib/ansible/modules/packaging/os/apt_repo.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+DOCUMENTATION = '''
+---
+module: apt_repo
+short_description: Manage APT repositories via apt-repo
+description:
+  - Manages APT repositories using apt-repo tool.
+  - See U(https://www.altlinux.org/Apt-repo) for details about apt-repo
+notes:
+    - This module works on ALT based distros.
```
Comment (gundalow) [inline lib/ansible/modules/packaging/os/apt_repo.py:23]:
```suggestion
    - This module works on ALT based distros.
    - Does NOT support checkmode, due to a limitation in apt-repo tool.
```

File: lib/ansible/modules/packaging/os/apt_repo.py (hunk 2 above)
Comment (gundalow) [reply lib/ansible/modules/packaging/os/apt_repo.py:114]: Makes sense, thank you

**Review by gundalow — COMMENTED** [review]
One minor comment then I think we can merge

File: lib/ansible/modules/packaging/os/apt_repo.py (hunk 2 above)
Comment (Obirvalger) [reply lib/ansible/modules/packaging/os/apt_repo.py:114]: You are welcome

**Review by gundalow — APPROVED** [review]

**Comment by gundalow** [comment]
Merged into `devel` for release in Ansible 2.8

**Pull request merged**